pkg/extensions/tbtc/monitoring_lock.go

```diff
@@ -0,0 +1,55 @@
+package tbtc
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+)
+
+type monitoringLockKey string
+
+func newMonitoringLockKey(
+	depositAddress string,
+	monitoringName string,
+) monitoringLockKey {
+	return monitoringLockKey(fmt.Sprintf(
+		"%v-%v",
+		depositAddress,
+		strings.ReplaceAll(monitoringName, " ", ""),
+	))
+}
+
+type monitoringLock struct {
```

Comment:
Could thus be better served by directly using [`sync.Map`](https://golang.org/pkg/sync/#Map 
)?

Reply:
Yeah, good idea!